cfgrammar/src/lib/header.rs

```diff
@@ -14,21 +16,35 @@ use std::{error::Error, fmt};
 /// * An error during parsing the section.
 /// * An error resulting from a value in the section having an invalid value.
 #[derive(Debug, Clone)]
-pub struct HeaderError {
+pub struct HeaderError<T> {
     pub kind: HeaderErrorKind,
-    pub locations: Vec<Location>,
+    pub locations: Vec<T>,
 }
 
-impl Error for HeaderError {}
-impl fmt::Display for HeaderError {
+impl<T: fmt::Debug> Error for HeaderError<T> {}
+impl<T> fmt::Display for HeaderError<T> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "{}", self.kind)
     }
 }
 
-impl From<HeaderError> for ParserError {
-    fn from(e: HeaderError) -> ParserError {
-        ParserError::HeaderError(e)
+impl From<HeaderError<Span>> for YaccGrammarError {
+    fn from(e: HeaderError<Span>) -> YaccGrammarError {
+        YaccGrammarError {
+            kind: YaccGrammarErrorKind::Header(e.kind, e.spanskind()),
+            spans: e.locations,
+        }
+    }
+}
+
+// This is essentially a tuple that needs a newtype so we can implement `From` for it.
+// Thus we aren't worried about it being `pub`.
+#[derive(Debug, PartialEq)]
+pub struct HeaderValue<T>(pub T, pub Value<T>);
```

Comment:
Likely needs `#[doc(hidden)]`, and I should just go through the docs again, to ensure there isn't anything else I added that also needs it.

Reply:
I believe all the doc changes should be fixed across various commits,
this one in [7895675](https://github.com/softdevteam/grmtools/pull/554/commits/7895675b1309b354179855f1c1979d70c52d7518)